sandbox.go

```diff
@@ -263,6 +266,39 @@ func (sb *sandbox) updateGateway(ep *endpoint) error {
 	return nil
 }
 
+func (sb *sandbox) SetKey(basePath string) error {
+	var err error
+	if basePath == "" {
+		sb.Unlock()
+		return fmt.Errorf("invalid sandbox key")
+	}
+
+	sb.Lock()
+	if sb.osSbox != nil {
+		sb.Unlock()
+		return fmt.Errorf("failed to set sandbox key : already assigned")
+	}
+	endpoints := sb.endpoints
+
+	sb.osSbox, err = osl.GetSandboxForExternalKey(basePath, sb.Key())
+	sb.Unlock()
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if err != nil {
+			sb.osSbox = nil
+		}
+	}()
+
+	for _, ep := range endpoints {
+		if err = sb.populateNetworkResources(ep); err != nil {
```

Comment:
This would again add the endpoint to the heap. Just wondering whether it could be a problem.